### PR TITLE
handle name of container to be valid

### PR DESCRIPTION
**Pull request by vhvb1989**
In aspire, for a container resource like:

```
var rabbitMq = builder.AddRabbitMQContainer("EventBus");
```

azd will fail trying to use `EvenBus` as the name of the container when creating the ACA.

The change in this PR converts the name into a valid container name.

### CODE REVIEW

File: cli/azd/resources/apphost/templates/resources.bicept [hunk 1]
```diff
@@ -168,7 +168,7 @@ resource {{bicepName $name}} 'Microsoft.App/containerApps@2023-05-02-preview' =
       containers: [
         {
           image: '{{$value.Image}}'
-          name: '{{$name}}'
+          name: '{{containerAppName $name}}-${resourceToken}'
```
Comment (weikanglim) [inline cli/azd/resources/apphost/templates/resources.bicept:171]: The `containerAppName` which is bound to `scaffold.ContainerAppName` was meant for `Microsoft.ContainerApps`, not for the container within -- naming restrictions may be different.

Comment (weikanglim) [reply cli/azd/resources/apphost/templates/resources.bicept:171]: What are the character restrictions here? FYI, recently Aspire has started restricting resource identifiers to alphanumeric + dash, with no dash as the first or last character.

Comment (weikanglim) [inline cli/azd/resources/apphost/templates/resources.bicept:171]: We may want to be consistent across the board -- for `AddProject`, we currently do not transform the name either when generated in the container app manifest.

**Comment by vhvb1989** [comment]
This is not required, as the name of the containerapp needs to be deterministic before provision to be built in project's env

**Pull request closed without merging**